components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2111,6 +2111,28 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
+    async getPriceInformation(ctx: TraceContext, attributionId: string): Promise<string | undefined> {
+        const user = this.checkAndBlockUser("createStripeCustomerIfNeeded");
+        const attrId = AttributionId.parse(attributionId);
+        if (!attrId) {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attributionId '${attributionId}'`);
+        }
+        let team: Team | undefined;
+        if (attrId.kind === "team") {
+            team = await this.guardTeamOperation(attrId.teamId, "update");
+            await this.ensureStripeApiIsAllowed({ team });
+        } else {
+            if (attrId.userId !== user.id) {
+                throw new ResponseError(
+                    ErrorCodes.PERMISSION_DENIED,
+                    "Cannot create Stripe customer profile for another user",
+                );
+            }
+            await this.ensureStripeApiIsAllowed({ user });
+        }
+        return this.stripeService.getPriceInformation(attributionId);
```

Comment:
**Suggestion**: We Can follow the consistent approach (used above) to return err if Price Information is not available. (Would be easy to debug in future)

```suggestion
        const priceInformation = this.stripeService.getPriceInformation(attributionId);
        if (!priceInformation) {
            throw new ResponseError(
                ErrorCodes.NOT_FOUND,
                `No price information found for attributionId '${attributionId}'`,
            );
        }
        return priceInformation;
```